src/robot.js

```diff
@@ -21,11 +22,28 @@ class Robot {
   // Robots receive messages from a chat source (Campfire, irc, etc), and
   // dispatch them to matching listeners.
   //
-  // adapterPath -  A String of the path to built-in adapters (defaults to src/adapters)
-  // adapter     - A String of the adapter name.
-  // httpd       - A Boolean whether to enable the HTTP daemon.
-  // name        - A String of the robot name, defaults to Hubot.
-  constructor (adapterPath, adapter, httpd, name, alias) {
+  // @param {Object} params - parameters for building Robot
+  // @param {String} params.adapterPath -  A String of the path to built-in adapters (defaults to src/adapters)
+  // @param {String} params.adapter     - A String of the adapter name.
+  // @param {String} params.httpd       - A Boolean whether to enable the HTTP daemon.
+  // @param {String} params.name        - A String of the robot name, defaults to Hubot.
+  constructor (params) {
```

Comment:
some ~googling~ research suggestions `options` is a javascript thing to, ie http://www.codereadability.com/what-are-javascript-options-objects/